### DISCUSSION TITLE
Access violation while using Zoom Panel in Status Bar

### DISCUSSION THREAD

**Discussion by urmel1960** [WinEdt-Team/WinEdt-Support#27] [2.2 Issues, Bugs and Problems]
- portable Build: 20240319  (v. 11.2) - 64-bit
- Win 11 Enterprise
- if there is need for further information, please let me know

I tried the new behaviour as described:

````
- The Zoom (Font Size) Panel in the Status Bar now responds to:
    Ctrl+Shift+Click as Zoom In
    Ctrl+Click       as Zoom Out
    Click            as Reset Font Zoom
````

I do get an access violation:

<img width="337" alt="24-03-2024_11-44-43" src="https://github.com/WinEdt-Team/WinEdt-Support/assets/144102/a84aea21-4210-48c4-b106-3fa4b297e3ea">

It was not directly reproducible, but I hope it helps to see if there is an issue.

When the problem occurred, I had not yet made any adjustments to the delivered configuration.

**Comment by Alex-Simonic** [comment]
The address did not help locate the problem. 

Do you have many large softly wrapped documents opened?
Did you execute the command more than once in rapid succession?

I suspect this is a sync problem with documents being reformatted in the background after the font size changed and a new zoom executed before the old one was applied to all documents. I could not reproduce it but the sync problem would explain why it happens sometimes but not always. If you think that this is a plausible explanation and you can reproduce it again (but not always) try the following:

Make a local copy of `%B\Menus\View\Zoom.edt`. Replace the line:

`LoadConfig('%b\Zoom.ini');
`

with

```
  ProcessMessages;
  LoadConfig('%b\Zoom.ini');
  Wait(111); // Try a few values 99..999 to see how much might be needed
  ProcessMessages;
```

and see if this fixes the problem. I don't have any other ideas right now. Let us know what you find...

Another thing to check: does a particular document cause problems after a font change?

**Comment by urmel1960** [comment]
> Did you execute the command more than once in rapid succession?

Thanks for pointing that out, yes - I'm sure I did. 

I zoomed in and out several times in succession to find the maximum zoom levels

The test document I used is not very large and only has 800 lines. Soft wrapping is not very extensive as I am working on a 32 inch 4k monitor.

I tried a little more and was able to reproduce it after a few attempts. 

If the problem occurs, you should quit WinEDT and restart it, otherwise the error message will keep coming back if you change the zoom in any way - even if you then no longer execute the functions in quick succession. 

I'll keep your suggested work-around in mind in case it should still be necessary somehow. 

Since the procedure I have followed is not a common use case, the issue can certainly be closed.
